plugins/http_plugin/tests/unit_tests.cpp

```diff
@@ -642,13 +644,14 @@ BOOST_FIXTURE_TEST_CASE(bytes_in_flight, http_plugin_test_fixture) {
 
    //load up some more requests that exceed max
    send_4mb_requests(32u);
-   //make sure got to the point http threads had responses queued
-   std::this_thread::sleep_for(std::chrono::seconds(1));
    //now rip these connections out before the responses are completely sent
    connections.clear();
```

Comment:
If I understood your description right, this is the area that is problematic. Between this line and the line below the http plugin needs to have cleaned up all its ownstanding async_writes. Just closing the socket here does indeed send the FIN but the http threads need time to wake up and handle that error on the async_write to then clear out what's in flight.

I wonder if what might work is to `shutdown` each connection and then `async_read` on each one until getting confirmation the remote side disconnected via an error code.

Reply:
That would likely work. I added a check for `bytes_in_flight` since that is really what we want to test anyway. We want to verify that bytes in flight goes to 0 once connections are closed.